### PR TITLE
Upgrade to .NET 6

### PR DESCRIPTION
**Pull request by nojaf**
Hello, I've upgraded the SDK and some of the projects to dotnet 6:
- Everything `netcoreapp3.1` is now `net6.0`
- Removed the `Fake.DotNet.Cli` usage in the FAKE scripts. I'm hitting https://github.com/fsprojects/FAKE/issues/2722
- Update GitHub Action
- Update dotnet tools
- Tweaked a couple of `.fs` files that had mismatched parameter names with the signature file.

### CODE REVIEW

**Comment by nojaf** [comment]
@kkm000, @dsyme, @sergey-tihon I'm still quite interested in a new release.
Especially to have https://github.com/fsprojects/FsLexYacc/pull/154.

**Comment by sergey-tihon** [comment]
Current tools build with `<RollForward>Major</RollForward>` so they should work on all .NET version 3.1+.
Do you have any issues with using the latest tool from NuGet?

I do not mind (personally) do upgrade to .NET 6, just would be nice to confirm with @dsyme that it works for compiler repo.

**Comment by nojaf** [comment]
> Do you have any issues with using the latest tool from NuGet?

Nope, I used the library (netstandard). 
As .NET Core 3.1 [is no longer supported](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core), I figured to update to the latest LTS.

The F# compiler uses this tool as [net7.0](https://github.com/dotnet/fsharp/blob/main/buildtools/fslex/fslex.fsproj#L5) so I don't think there is any harm here.

File: build.fsx [hunk 1]
```diff
@@ -123,12 +123,9 @@ Target.create "Build" (fun _ ->
...
         for project in ["src/FsLex/fslex.fsproj"; "src/FsYacc/fsyacc.fsproj"] do
-            DotNet.publish (fun opts -> { 
-                opts with 
-                    Common = { opts.Common with CustomParams = Some "/v:n" }
-                    Configuration = DotNet.BuildConfiguration.Release
-                    Framework = Some framework 
-            }) project
+            CreateProcess.fromRawCommandLine "dotnet" $"publish {project} -c Release /v:n -f {framework}"
+            |> Proc.run
+            |> ignore
```
Comment (sergey-tihon) [inline build.fsx:128]: Are you sure that you can just ignore result of execution? What if process fail (build, test, etc)? I believe current code just ignore exit code and build will pass.

I think we should check exit code and fail if exit code is not zero, like this https://github.com/fsprojects/SwaggerProvider/blob/master/build.fsx#L78-L84

**Comment by dsyme** [comment]
Great, thanks!

**Pull request merged**